src/pkgcore/ebuild/repo_objs.py

```diff
@@ -170,6 +171,217 @@ def __init__(self, metadata_xml, manifest):
+        else:
+            res = self.email
+        if self.url is not None:
+            return f'{res} ({self.url})'
+        return res
+
+    @property
+    def leads(self):
+        """Project lead(s), if any."""
+        return tuple(m for m in self.members if m.is_lead)
+
+    @property
+    def recursive_members(self):
+        """All project members, including members inherited from subprojects."""
+        # collect all subprojects from which to inherit, recursively
+        subprojects = list(sp for sp in self.subprojects if sp.inherit_members)
+        subproject_emails = set(sp.email for sp in subprojects)
+        i = 0
+        while i < len(subprojects):
+            for sp in subprojects[i].subprojects:
+                if sp.inherit_members and sp.email not in subproject_emails:
+                    subprojects.append(sp)
+                    subproject_emails.add(sp.email)
+            i += 1
+
+        members = list(self.members)
+        emails = set(m.email for m in self.members)
+        for sp in subprojects:
+            if sp.inherit_members:
+                for m in sp.members:
+                    # avoid duplicating people
+                    if m.email not in emails:
+                        # drop lead bit
+                        m = ProjectMember(email=m.email,
+                                          name=m.name,
+                                          role=m.role)
+                        members.append(m)
+                        emails.add(m.email)
+        return tuple(members)
+
+
+class ProjectsXml(object):
+    """projects.xml parsed results
+
+    Attributes are set to -1 if unloaded, None if no entry, or the value
+    if loaded.
+    """
+
+    __slots__ = (
+        "__weakref__", "_projects", "_source"
+    )
+
+    def __init__(self, source):
+        self._source = source
+
+    @property
+    def projects(self):
+        if self._source is not None:
+            self._parse_xml()
+        return self._projects
+
+    def _parse_xml(self, source=None):
+        if source is None:
+            source = self._source.bytes_fileobj()
+        try:
+            tree = etree.parse(source)
+        except etree.XMLSyntaxError as pe:
+            self._projects = mappings.ImmutableDict()
+            self._source = None
+            logger.error(f"failed parsing projects.xml: {pe}")
+            return
+
+        projects = {}
+        for p in tree.findall("project"):
+            kwargs = {}
+            for k in ('email', 'name', 'url', 'description'):
+                kwargs[k] = p.findtext(k)
+
+            members = []
+            for m in p.findall('member'):
+                m_kwargs = {}
+                for k in ('email', 'name', 'role'):
+                    m_kwargs[k] = m.findtext(k)
+                m_kwargs['is_lead'] = m.get('is-lead', '') == '1'
+                members.append(ProjectMember(**m_kwargs))
+            kwargs['members'] = tuple(members)
```

Comment:
I'd move the tuple() to the related object instantiation so the related object attr is always stored as an attribute even when creating a singular object.